npreadtext/_readers.py

```diff
@@ -32,6 +32,65 @@ def _preprocess_comments(iterable, comments, encoding):
         yield line
 
 
+_CHUNK_SIZE = 50000
+
+def _read_in_chunks(file, *,
+        delimiter, comment, quote, decimal, sci, imaginary_unit,
+        usecols, skiprows, max_rows, converters, ndmin, unpack,
+        dtype, encoding):
+    """ Function "implementing" string-length and time-unit from string
+    discovery.  The core C-functionality does not implement this yet
+    (especially for converters).
+    """
+    requested_dtype = dtype.char
+    dtype = object
+
+    if isinstance(file, str):
+        data = np.lib._datasource.open(file, 'rt', encoding=encoding)
+        opened = True
+    else:
+        data = file
+        opened = False
```

Comment:
It would be better to reorder the main reader functionality a bit to allow sharing the logic better and not actualling `read` here, but rather the C-reader directly.